quic/s2n-quic-core/src/recovery/hybrid_slow_start.rs

```diff
@@ -94,29 +135,73 @@ impl HybridSlowStart {
+                self.css_count += 1;
+                if cur_min_rtt < self.css_baseline_min_rtt {
+                    // resume slow start
+                    self.css_threshold = self.threshold;
+                    self.ss_growth_divisor = 1.0;
+                    self.css_count = 0;
+                }
+                if self.css_count >= CSS_ROUNDS {
+                    // exit slow start phase
+                    self.threshold = congestion_window;
+                    self.css_threshold = f32::MAX;
+                    self.ss_growth_divisor = 1.0;
+                }
+            } else {
+                let threshold = last_min_rtt / THRESHOLD_DIVIDEND;
+                // Clamp n to the min and max thresholds
+                let threshold = threshold.min(MAX_DELAY_THRESHOLD).max(MIN_DELAY_THRESHOLD);
+                let delay_increase_is_over_threshold = cur_min_rtt >= last_min_rtt + threshold;
+                let congestion_window_is_above_minimum = congestion_window >= self.low_ssthresh();
+
+                if self.use_hystart_plus_plus {
+                    // if delay is beyond threshold, go into css phase
+                    if delay_increase_is_over_threshold {
+                        self.css_threshold = congestion_window;
+                        self.css_baseline_min_rtt = cur_min_rtt;
+                        self.ss_growth_divisor = CSS_GROWTH_DIVISOR;
+                        self.css_count = 0;
+                    }
+                } else if delay_increase_is_over_threshold && congestion_window_is_above_minimum {
+                    self.threshold = congestion_window;
+                }
             }
         }
     }
 
+    /// return cwnd increment during slow start phase
+    /// should be called from on_packet_ack
+    pub fn cwnd_increment(&self, sent_bytes: usize) -> f32 {
+        if cfg!(debug_assertions) {
+            if !self.use_hystart_plus_plus {
+                assert_eq!(self.ss_growth_divisor, 1.0);
+            }
+        }
+        (sent_bytes as f32) / self.ss_growth_divisor
+    }
+
     /// Called when a congestion event is experienced. Sets the
     /// slow start threshold to the minimum of the Hybrid Slow Start threshold
     /// and the given congestion window. This will ensure we exit slow start
     /// early enough to avoid further congestion.
     pub fn on_congestion_event(&mut self, ssthresh: f32) {
         self.threshold = self.threshold.min(ssthresh).max(self.low_ssthresh());
+        self.ss_growth_divisor = 1.0;
+        self.css_threshold = f32::MAX;
     }
 
     fn low_ssthresh(&self) -> f32 {
         (LOW_SSTHRESH * self.max_datagram_size) as f32
     }
+
+    fn get_hystart_parameter() -> bool {
+        use once_cell::sync::OnceCell;
+        static USE_HYSTART_PP: OnceCell<bool> = OnceCell::new();
+        #[cfg(feature = "std")]
+        *USE_HYSTART_PP.get_or_init(|| std::env::var(USE_HYSTART_PLUS_PLUS).is_ok())
```

Comment:
I think you would have to have some boolean variable that is always returned by this function, otherwise I don't think it will compile without the standard library